crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -99,30 +110,44 @@ async fn handle_accepted<Client, Pool>(
     }
 }
 
-impl<Client, Pool> std::fmt::Debug for EthPubSub<Client, Pool> {
+impl<Client, Pool, Events> std::fmt::Debug for EthPubSub<Client, Pool, Events> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("EthPubSub").finish_non_exhaustive()
     }
 }
 
 /// Container type `EthPubSub`
 #[derive(Clone)]
-struct EthPubSubInner<Client, Pool> {
+struct EthPubSubInner<Client, Pool, Events> {
     /// The transaction pool.
     pool: Pool,
     /// The client that can interact with the chain.
     client: Client,
+    /// A type that allows to create new event subscriptions,
+    chain_events: Events,
 }
 
 // == impl EthPubSubInner ===
 
-impl<Client, Pool> EthPubSubInner<Client, Pool>
+impl<Client, Pool, Events> EthPubSubInner<Client, Pool, Events>
 where
-    Client: BlockProvider + 'static,
     Pool: TransactionPool + 'static,
 {
     /// Returns a stream that yields all transactions emitted by the txpool.
     fn into_pending_transaction_stream(self) -> impl Stream<Item = TxHash> {
         ReceiverStream::new(self.pool.pending_transactions_listener())
     }
 }
+
+impl<Client, Pool, Events> EthPubSubInner<Client, Pool, Events>
+where
+    Client: BlockProvider + 'static,
+    Events: ChainEventSubscriptions + 'static,
+{
+    /// Returns a stream that yields all new RPC blocks.
+    fn into_new_headers_stream(self) -> impl Stream<Item = Header> {
+        UnboundedReceiverStream::new(self.chain_events.subscribe_new_blocks()).map(|new_block| {
+            Header::from_primitive_with_hash(new_block.header.as_ref().clone(), new_block.hash)
+        })
+    }
+}
```

Comment:
we should also expose the full versions of these as non-standard calls, for Transaction and Block instead of TxHash BlockHash